proof/refine/ARM/Ipc_R.thy

```diff
@@ -2335,289 +2344,831 @@ crunches doIPCTransfer, possibleSwitchTo
-                  apply (simp add: if_apply_def2)
-                  apply (wp hoare_drop_imps)
-                  apply (simp add: if_apply_def2)
-                apply ((wp sts_cur_tcb set_thread_state_runnable_weak_valid_sched_action sts_st_tcb_at_cases |
-                           simp add: if_apply_def2 split del: if_split)+)[1]
-               apply (wp setThreadState_valid_queues' sts_valid_queues sts_weak_sch_act_wf
-                         sts_cur_tcb' setThreadState_tcb' sts_st_tcb_at'_cases)
-              apply (simp add: valid_tcb_state_def pred_conj_def)
-              apply ((wp hoare_drop_imps do_ipc_transfer_tcb_caps  weak_valid_sched_action_lift
-                     | clarsimp simp:is_cap_simps)+)[1]
-             apply (simp add: valid_tcb_state'_def pred_conj_def)
-             apply (strengthen sch_act_wf_weak)
-             apply (wp weak_sch_act_wf_lift_linear hoare_drop_imps)
-            apply (wp gts_st_tcb_at)+
-          apply (simp add: ep_relation_def split: list.split)
-         apply (simp add: pred_conj_def cong: conj_cong)
-         apply (wp hoare_post_taut)
-        apply simp
-        apply (wp weak_sch_act_wf_lift_linear set_ep_valid_objs' setEndpoint_valid_mdb')
-       apply (clarsimp simp add: invs_def valid_state_def
-                                 valid_pspace_def ep_redux_simps ep_redux_simps'
-                                 st_tcb_at_tcb_at
-                           cong: list.case_cong)
-       apply (clarsimp simp: valid_ep_def)
-       apply (drule(1) sym_refs_obj_atD[where P="\<lambda>ob. ob = e" for e])
-       apply (clarsimp simp: st_tcb_at_refs_of_rev st_tcb_at_reply_cap_valid
-                             st_tcb_at_caller_cap_null)
-       apply (fastforce simp: st_tcb_def2 valid_sched_def valid_sched_action_def)
-      subgoal by (auto simp: valid_ep'_def
-                      split: list.split;
-                  clarsimp simp: invs'_def valid_state'_def)
-     apply wp+
-   apply (clarsimp simp: ep_at_def2)+
-  done *)
-qed
-
-end
-
-global_interpretation setMessageInfo: typ_at_all_props' "setMessageInfo t info"
-  by typ_at_props'
-
-context begin interpretation Arch . (*FIXME: arch_split*)
+    apply (rule corres_split_eqr)
+       apply (rule corres_split_eqr)
+          apply (rule corres_if, simp)
+           apply (rule corres_if, simp)
+            apply (rule corres_trivial, simp)
+           apply simp
+          apply (rule corres_trivial, simp)
+         apply (rule threadget_corres, simp add: tcb_relation_def)
+        apply (wp thread_get_wp)
+       apply (wp threadGet_wp)
+      apply (rule threadget_corres, simp add: tcb_relation_def)
+     apply (wp thread_get_wp)
+    apply (wpsimp wp: threadGet_wp)
+   apply (fastforce simp: projectKO_eq projectKO_tcb obj_at'_def)+
+  done
 
-crunches cancel_ipc
-  for cur[wp]: "cur_tcb"
-  and ntfn_at[wp]: "ntfn_at t"
-  (wp: select_wp crunch_wps simp: crunch_simps ignore: set_object)
+lemma tcbEPAppend_corres:
```

Comment:
Did you do the corres rename in this PR? I'm curious since it's not caught in my PR, unless this is an MCS specific operation.

Reply:
I did this rename in my PR. Not sure whether it is MCS specific.